Server/modules/luaModules/src/CLuaModulesModule.cpp

```diff
@@ -0,0 +1,104 @@
+#include "StdInc.h"
+#include <sdk/SharedUtil.File.h>
+
+void CLuaModulesModule::Load()
+{
+    m_pLuaModuleManager = new CLuaModuleManager(m_pModuleInterface);
+    
+}
+
+void CLuaModulesModule::UnLoad()
+{
+}
+
+void CLuaModulesModule::DoPulse()
+{
+    m_pLuaModuleManager->DoPulse();
+}
+
+void CLuaModulesModule::ResourceStarted(IResource* pResource)
+{
+}
+
+void CLuaModulesModule::ResourceStopped(IResource* pResource)
+{
+    m_pLuaModuleManager->ResourceStopped(pResource);
+}
+
+void CLuaModulesModule::ResourceStopping(IResource* pResource)
+{
+    m_pLuaModuleManager->ResourceStopping(pResource);
+}
+
+int CLuaModulesModule::LoadModule(const char* szShortFileName, const char* szFileName, bool bLateLoad)
+{
+    return m_pLuaModuleManager->LoadModule(szShortFileName, szFileName, bLateLoad);
+}
+
+int CLuaModulesModule::UnloadModule(const char* szShortFileName)
+{
+    return m_pLuaModuleManager->UnloadModule(szShortFileName);
+}
+
+int CLuaModulesModule::ReloadModule(const char* szShortFileName, const char* szFileName, bool bLateLoad)
+{
+    return m_pLuaModuleManager->ReloadModule(szShortFileName, szFileName, bLateLoad);
+}
+
+ILuaModule* CLuaModulesModule::GetModuleByName(const char* szName)
+{
+    std::vector<CLuaModule*> luaModules = m_pLuaModuleManager->GetLoadedModules();
+    for (const auto luaModule : luaModules)
+    {
+        if (luaModule->_GetName() == szName)
+        {
+            return luaModule;
+        }
+    }
+    return nullptr;
+}
+
+std::vector<ILuaModule*> CLuaModulesModule::GetModules()
+{
+    std::vector<CLuaModule*> luaModules = m_pLuaModuleManager->GetLoadedModules();
+    std::vector<ILuaModule*> modules(luaModules.size());
```

Comment:
Could just:
```cpp
    const auto& modules = m_pLuaModuleManager->GetLoadedModules();
    return {modules.begin(), modules.end()};
```
 instead of this mess.